opslevel/resource_opslevel_filter.go

```diff
@@ -48,39 +54,75 @@ type filterPredicate struct {
+var filterPredicateType = map[string]attr.Type{
+	"case_insensitive": types.BoolType,
+	"case_sensitive":   types.BoolType,
+	"key":              types.StringType,
+	"key_data":         types.StringType,
+	"type":             types.StringType,
+	"value":            types.StringType,
+}
+
+func (fp filterPredicateModel) Validate() error {
+	opslevelFilterPredicate := opslevel.FilterPredicate{
+		CaseSensitive: fp.CaseSensitive.ValueBoolPointer(),
+		Key:           opslevel.PredicateKeyEnum(fp.Key.ValueString()),
+		KeyData:       fp.KeyData.ValueString(),
+		Type:          opslevel.PredicateTypeEnum(fp.Type.ValueString()),
+		Value:         fp.Value.ValueString(),
 	}
-	if predicate.CaseSensitive != nil {
-		isCaseSensitive := *predicate.CaseSensitive
-		if isCaseSensitive {
-			convertedFilterPredicate.CaseSensitive = types.BoolValue(true)
-			convertedFilterPredicate.CaseInsensitive = types.BoolValue(false)
-		} else {
-			convertedFilterPredicate.CaseSensitive = types.BoolValue(false)
-			convertedFilterPredicate.CaseInsensitive = types.BoolValue(true)
-		}
+	if opslevelFilterPredicate.CaseSensitive == nil && !fp.CaseInsensitive.IsNull() {
+		opslevelFilterPredicate.CaseSensitive = fp.CaseInsensitive.ValueBoolPointer()
 	}
-	return convertedFilterPredicate
+	return opslevelFilterPredicate.Validate()
 }
 
-func NewFilterResourceModel(filter opslevel.Filter) FilterResourceModel {
-	filterPredicates := []filterPredicate{}
-	for _, predicate := range filter.Predicates {
-		filterPredicates = append(filterPredicates, convertPredicate(predicate))
+func NewFilterResourceModel(ctx context.Context, filter opslevel.Filter, givenModel FilterResourceModel) (FilterResourceModel, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	var filterPredicateAttrs []attr.Value
+	var filterPredicatesListValue basetypes.ListValue
+	var givenPredicateModels []filterPredicateModel
+
+	// Convert predicates from plan to slice of models
+	givenModel.Predicate.ElementsAs(ctx, &givenPredicateModels, false)
+
+	for _, opslevelPredicate := range filter.Predicates {
+		predicateObj := OpslevelFilterPredicateToObjectValue(nil, &opslevelPredicate)
+		attrs := predicateObj.Attributes()
+
+		// find predicate from plan that matches predicateObj from API
+		foundPlanPredModel, extractDiags := ExtractFilterPredicateModel(ctx, attrs, givenPredicateModels)
+		diags.Append(extractDiags...)
+		if diags.HasError() {
+			return FilterResourceModel{}, diags
+		}
+
+		if !foundPlanPredModel.CaseSensitive.IsNull() && !foundPlanPredModel.CaseSensitive.IsUnknown() {
+			attrs["case_sensitive"] = types.BoolValue(*opslevelPredicate.CaseSensitive)
+		} else {
+			attrs["case_sensitive"] = types.BoolNull()
+		}
+		if !foundPlanPredModel.CaseInsensitive.IsNull() && !foundPlanPredModel.CaseInsensitive.IsUnknown() {
+			attrs["case_insensitive"] = types.BoolValue(*opslevelPredicate.CaseSensitive)
+		} else {
+			attrs["case_insensitive"] = types.BoolNull()
```

Comment:
😬 - i get why but maybe this would have been a good case for a breaking change and 1.1.0 

The reason it was 2 fields before was because we coudn't tell the difference between, false and null.  Now we can so it could go back to 1 field.

Reply:
Tellllllll me about it 😵 . But I wanted to follow the [Terraform deprecation recommendations](https://github.com/OpsLevel/terraform-provider-opslevel/pull/376) so maybe the next release with be a minor release